hawc/apps/animalv2/models.py

```diff
@@ -0,0 +1,576 @@
+    # Or just build the UI to show options from a list but also let user type custom ones
+    # (in that case just store it as a string)? leave as string for now.
+    effect_modifier_timing = models.CharField(max_length=128, blank=True)
+    effect_modifier_reference = models.CharField(max_length=128, blank=True)
+    effect_modifier_anatomical = models.CharField(max_length=128, blank=True)
+    effect_modifier_location = models.CharField(max_length=128, blank=True)
+    comments = models.TextField(blank=True, help_text="TODO")
+
+    def __str__(self):
+        return self.name
+
+    def get_assessment(self):
+        return self.experiment.get_assessment()
+
+    def get_study(self):
+        return self.experiment.get_study()
+
+
+class AnimalEndpointTimepoint(models.Model):
+    objects = managers.AnimalEndpointTimepointManager()
+
+    # TODO
+    TEXT_CLEANUP_FIELDS = ()
+
+    endpoint = models.ForeignKey(
+        AnimalEndpoint, on_delete=models.CASCADE, related_name="v2_timepoints"
+    )
+    observation_time = models.FloatField(
+        blank=True,
+        null=True,
+        verbose_name="Observation timepoint",
+        help_text="Numeric value of the time an observation was reported; "
+        "optional, should be recorded if the same effect was measured multiple times.",
+    )
+    observation_time_units = models.PositiveSmallIntegerField(
+        default=constants.ObservationTimeUnits.NR, choices=constants.ObservationTimeUnits.choices
+    )
+    observation_time_text = models.CharField(
+        max_length=64,
+        blank=True,
+        help_text='Text for reported observation time (ex: "60-90 PND")',
+    )
+    comments = models.TextField(blank=True, help_text="TODO")
+    created = models.DateTimeField(auto_now_add=True)
+    last_updated = models.DateTimeField(auto_now=True)
+
+    def get_assessment(self):
+        return self.endpoint.get_assessment()
+
+    def get_study(self):
+        return self.endpoint.get_study()
+
+    # TODO - flat_complete_header_row / flat_complete_data_row
+
+
+class DataExtraction(models.Model):
+    objects = managers.DataExtractionManager()
+
+    experiment = models.ForeignKey(
+        Experiment, on_delete=models.CASCADE, related_name="v2_data_extractions"
+    )
+    endpoint = models.ForeignKey(
+        AnimalEndpoint, on_delete=models.CASCADE, related_name="v2_data_extractions"
+    )
+    treatment = models.ForeignKey(
+        Treatment, on_delete=models.CASCADE, related_name="v2_data_extractions"
+    )
+
+    # specifically, DoseGroup.dose?
+    dose = models.ForeignKey(
+        DoseGroup, on_delete=models.CASCADE, related_name="v2_data_extractions"
+    )
+    observation_timepoint = models.ForeignKey(
+        AnimalEndpointTimepoint, on_delete=models.CASCADE, related_name="v2_data_extractions"
+    )
+
+    # specific fields
+    n = models.PositiveSmallIntegerField(blank=True, null=True, validators=[MinValueValidator(0)])
```

Comment:
i think many of these fields may only belong on the `DoseResponseGroupLevelData` - dose, n, variance, treatment_related_effect

Reply:
I'll circle back with Vidhi and Pam. That was my thought too, that it seemed very odd to have these in both locations, latest guidance I got from them was that it very explicitly should be that way. But like I said, seems weird to me and sounds like it seems weird to you. We'll discuss.

Reply:
ok think we have this sorted now , stuff has been moved down to the group level data instead of the parent extraction object.